com.unity.shadergraph/Editor/Generation/Collections/AssetCollection.cs

```diff
@@ -0,0 +1,47 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace UnityEditor.ShaderGraph
+{
+    // this class is used to track asset dependencies in shadergraphs and subgraphs
+    // that is: it tracks files that the shadergraph or subgraph must access to generate the shader
+    // this data is used to automatically re-import the shadergraphs or subgraphs when any of the tracked files change
+    public class AssetCollection
```

Comment:
Is it expected for this class to be public? If so, you need to add the documentation

Reply:
good point, I can probably make it internal for now -- it will be made public when we make Targets public.